client/src/controllers/UpgradeController.ts

```diff
@@ -0,0 +1,84 @@
+import { Controller } from '@hotwired/stimulus';
+import { VersionNumber, VersionDeltaType } from '../utils/version';
+
+export class UpgradeController extends Controller<HTMLElement> {
+  static classes = ['hidden'];
+  static targets = ['latestVersion', 'link'];
+  static values = {
+    currentVersion: String,
+    ltsOnly: { default: false, type: Boolean },
+    url: { default: 'https://releases.wagtail.org/latest.txt', type: String },
+  };
+
+  currentVersionValue: string;
+  hiddenClass: string;
+  ltsOnlyValue: any;
+  latestVersionTarget: HTMLElement;
+  linkTarget: HTMLElement;
+  urlValue: string;
+
+  connect() {
+    this.checkVersion();
+  }
+
+  checkVersion() {
+    const releasesUrl = this.urlValue;
+    const currentVersion = new VersionNumber(this.currentVersionValue);
+    const showLTSOnly = this.ltsOnlyValue;
+    fetch(releasesUrl, {
+      referrerPolicy: 'strict-origin-when-cross-origin',
+    })
+      .then((response) => {
+        if (response.status !== 200) {
+          // eslint-disable-next-line no-console
+          console.error(
+            `Unexpected response from ${releasesUrl}. Status: ${response.status}`,
+          );
+          return false;
+        }
+        return response.json();
+      })
+      .then((payload) => {
+        let data = payload;
+        if (data && data.lts && showLTSOnly) {
+          data = data.lts;
+        }
+
+        if (data && data.version) {
+          const latestVersion = new VersionNumber(data.version);
+          const versionDelta = currentVersion.howMuchBehind(latestVersion);
+
+          let releaseNotesUrl = null;
+          if (!versionDelta) {
+            return;
+          }
+          if (
```

Comment:
It might be good to pull out this block (that writes the version) to a seperate method. But not critical

Reply:
Is it only line 52 to 54? That checks for when the current version isn't behind?
Or the second if statement is included

Reply:
All good leave as is maybe